services/src/main/java/org/keycloak/protocol/oidc/grants/device/endpoints/DeviceEndpoint.java

```diff
@@ -111,6 +111,11 @@ public Response handleDeviceRequest() {
         AuthorizationEndpointRequest request = AuthorizationEndpointRequestParserProcessor.parseRequest(event, session, client,
             httpRequest.getDecodedFormParameters());
 
+        if(request.getInvalidRequestMessage() != null) {
+            throw new ErrorResponseException(OAuthErrorException.INVALID_GRANT,
+                request.getInvalidRequestMessage(), Response.Status.BAD_REQUEST);
```

Comment:
How about adding `event.error(Errors.INVALID_REQUEST);` the same as:
https://github.com/keycloak/keycloak/blob/820ab52dce148eae0d51ba194ba9248bf83a69c6/services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpointChecker.java#L201-L203